### PR TITLE
Added missing robot state update to iterative spline parameterization

### PR DESCRIPTION
**Pull request by Martin-Oehler**
### Description
The iterative spline parameterization is missing an update to the robot states of the trajectory after iterating over each point. These leads to lots of warnings of the following kind:
```
[ WARN] [1547043223.679662634, 8.816000000] [/move_group]: Returning dirty collision body transforms
```
This PR adds the missing `update()` call (just one line).

I did not auto-format with clang as this changed lines unrelated to my PR. 

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Document API changes relevant to the user in the moveit/MIGRATION.md notes
- [ ] Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Decide if this should be cherry-picked to other current ROS branches
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by Martin-Oehler** [comment]
For some reason, the trajectory is first written to an internal data structure before doing the time parameterization. In this final step, the result is written back to the `robot_trajectory::RobotTrajectory` object.

**Comment by rhaschke** [comment]
Yes, I have seen this. The question is why? The joint positions shouldn't be modified, as this might invalidate the trajectory.

**Comment by Martin-Oehler** [comment]
Without looking further into it, I have only two guesses:
1. Performance: The data is restructured, maybe for faster access
2. The code has been copied from somewhere else and this was the fastest way to make it run.

**Comment by v4hn** [comment]
Updating all states can be quite a bit of unnecessary computation here and I would prefer to avoid it where possible.

Looking at the code, it [conditionally includes two new waypoints if `add_points` is set to true](https://github.com/ros-planning/moveit/blob/34570a12c17b6b7fca3d58131a0a07189d0164d9/moveit_core/trajectory_processing/src/iterative_spline_parameterization.cpp#L133) and the code only ever modifies the positions of these two points.

So how about we skip the [`setVariablePosition`](https://github.com/ros-planning/moveit/blob/34570a12c17b6b7fca3d58131a0a07189d0164d9/moveit_core/trajectory_processing/src/iterative_spline_parameterization.cpp#L343) calls and instead do them for the two new points only if `add_points_` is true (and afterwards `update` them too?

**Comment by rhaschke** [comment]
Thanks for looking deeper into this, @v4hn. I would agree to your proposal.

**Comment by Martin-Oehler** [comment]
I agree as well and will update my PR accordingly (or should I open a new one?).

**Comment by rhaschke** [comment]
Updating should be fine. You can also squash commits to cleanup history ;-)

**Comment by Martin-Oehler** [comment]
I implemented the requested changes.

File: moveit_core/trajectory_processing/src/iterative_spline_parameterization.cpp [hunk 1]
```diff
@@ -340,10 +340,19 @@ bool IterativeSplineParameterization::computeTimeStamps(robot_trajectory::RobotT
   {
     for (unsigned int j = 0; j < num_joints; j++)
     {
-      trajectory.getWayPointPtr(i)->setVariablePosition(idx[j], t2[j].positions[i]);
       trajectory.getWayPointPtr(i)->setVariableVelocity(idx[j], t2[j].velocities[i]);
       trajectory.getWayPointPtr(i)->setVariableAcceleration(idx[j], t2[j].accelerations[i]);
     }
+
+    // Only update position of additional points
```
Comment (rhaschke) [inline moveit_core/trajectory_processing/src/iterative_spline_parameterization.cpp]:
```suggestion
    // Only update position of additionally inserted points (at second and next-to-last position)
```

**Pull request merged**